src/data.js

```diff
@@ -1,57 +1,49 @@
 // import data from './data/pokemon/pokemon.js';
-
-// const pokemons {filterType,sortbyNameAZ,}
 export function filterType(allPokemon, elementPokemon) {
-  let filtered = [];
-      for (let i = 0; i < allPokemon.length; i++){
-          for (let e = 0; e < allPokemon[i].type.length; e++) {
-              if(allPokemon[i].type[e] === elementPokemon){
-                  filtered.push(allPokemon[i]);
-              }
-          }
-      }
+  const filtered = [];
+  for (let i = 0; i < allPokemon.length; i++) {
+    for (let e = 0; e < allPokemon[i].type.length; e++) {
+      if (allPokemon[i].type[e] === elementPokemon) {
```

Comment:
Muy buen approach! Te recomiendo buscar la manera de que no sean for anidados, así tu rendimiento será mucho mejor!